lib/base16.rb

```diff
@@ -1,9 +1,11 @@
+# See https://www.rfc-editor.org/rfc/rfc4648#section-8
 class Base16
-
-  # The IRS has requested data be encoded this way. Loosely emulate the Base64 class.
-
   def self.encode16(str)
```

Comment:
I think it will matter more in context as well, but another option is for us to shell out to `xxd`. I think once we start experimenting with big encrypting, we might just write the raw data to a file and shell out to `xxd | gzip` or something and I think that will be faster than doing things in ruby